server.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+
+const express = require('express');
+const cors = require('cors');
+
+const app = express();
+
+app.use(cors());
+
+require('dotenv').config();
+
+app.get('/location', (request, response) => {
+  try{
+    let searchQuery = request.query.data;
+    const geoDataResults = require('./data/geo.json');
+  
+
+    const location = new Location(searchQuery, geoDataResults);
+    response.status(200).send(location);
+
+  }
+  catch(err){
+    response.status(500).send("Sorry, something went wrong")
+    console.error(err);
+  }
+});
+
+app.get('/weather', (request, response) => {
+  try {
+    // const searchQuery = request.query.data;
+    const darkSkyDataResults = require('./data/darksky.json');
+    const weathers = darkSkyDataResults.daily.data.map(el => {
+      const forecast = el.summary;
+      const date = new Date(el.time);
+      
+      let weekday;
+      let month;
+
+      // Get weekday
+      switch (date.getDay()) {
+      case 0:
+        weekday = 'Sun';
+        break;
+      case 1:
+        weekday = 'Mon';
+        break;
+      case 2:
+        weekday = 'Tue';
+        break;
+      case 3:
+        weekday = 'Wed';
+        break;
+      case 4:
+        weekday = 'Thu';
+        break;
+      case 5:
+        weekday = 'Fri';
+        break;
+      case 6:
+        weekday = 'Sat';
+        break;
+      default:
+        break;
+      }
+      // Get month
+      switch (date.getMonth()) {
+      case 0:
+        month = 'Jan';
+        break;
+      case 1:
+        month = 'Feb';
+        break;
+      case 2:
+        month = 'Mar';
+        break;
+      case 3:
+        month = 'Apr';
+        break;
+      case 4:
+        month = 'May';
+        break;
+      case 5:
+        month = 'Jun';
+        break;
+      case 6:
+        month = 'Jul';
+        break;
+      case 7:
+        month = 'Aug';
+        break;
+      case 8:
+        month = 'Sep';
+        break;
+      case 9:
+        month = 'Oct';
+        break;
+      case 10:
+        month = 'Nov';
+        break;
+      case 11:
+        month = 'Dec';
+        break;
+      }
+
+      const time = `${weekday} ${month} ${date.getDate()}, ${date.getFullYear()}`;
```

Comment:
While I haven't checked closely for minor differences, it seems your use of the switch cases and parsing out the date information is probably exactly what we expect for this string. Good job catching some of the pitfalls of working with those date values. That said, it would be better to leverage the Date.toString() method in JavaScript so you aren't writing duplicate code and we know for certain we have the string formatted as expected.